src/main/java/org/broadinstitute/hellbender/utils/pairhmm/VectorLoglessPairHMM.java

```diff
@@ -0,0 +1,287 @@
+    @Override
+    public void computeLog10Likelihoods(final LikelihoodMatrix<Haplotype> logLikelihoods,
+                                        final List<GATKRead> processedReads,
+                                        final Map<GATKRead, byte[]> gcp) {
+        if (processedReads.isEmpty())
+            return;
+        if (doProfiling)
+            startTime = System.nanoTime();
+        int readListSize = processedReads.size();
+        int numHaplotypes = logLikelihoods.numberOfAlleles();
+        JNIReadDataHolderClass[] readDataArray = new JNIReadDataHolderClass[readListSize];
+        int idx = 0;
+        for (GATKRead read : processedReads) {
+            readDataArray[idx] = new JNIReadDataHolderClass();
+            readDataArray[idx].readBases = read.getBases();
+            readDataArray[idx].readQuals = read.getBaseQualities();
+            readDataArray[idx].insertionGOP = ReadUtils.getBaseInsertionQualities(read);
+            readDataArray[idx].deletionGOP = ReadUtils.getBaseDeletionQualities(read);
+            readDataArray[idx].overallGCP = gcp.get(read);
+            ++idx;
+        }
+
+        mLogLikelihoodArray = new double[readListSize * numHaplotypes];      //to store results
+        if (doProfiling)
+            threadLocalSetupTimeDiff = (System.nanoTime() - startTime);
+        //for(reads)
+        //   for(haplotypes)
+        //       compute_full_prob()
+        jniComputeLikelihoods(readListSize, numHaplotypes, readDataArray, mHaplotypeDataArray, mLogLikelihoodArray, 12);
+
+        int readIdx = 0;
+        for (int r = 0; r < readListSize; r++) {
+            int hapIdx = 0;
+            for (final Haplotype haplotype : logLikelihoods.alleles()) {
+
+                //Since the order of haplotypes in the List<Haplotype> and alleleHaplotypeMap is different,
+                //get idx of current haplotype in the list and use this idx to get the right likelihoodValue
+                final int idxInsideHaplotypeList = haplotypeToHaplotypeListIdxMap.get(haplotype);
+                logLikelihoods.set(hapIdx, r, mLogLikelihoodArray[readIdx + idxInsideHaplotypeList]);
+                ++hapIdx;
+            }
+            readIdx += numHaplotypes;
+        }
+        if (doProfiling) {
+            threadLocalPairHMMComputeTimeDiff = (System.nanoTime() - startTime);
+            //synchronized(doProfiling)
+            {
+                pairHMMComputeTime += threadLocalPairHMMComputeTimeDiff;
+                pairHMMSetupTime += threadLocalSetupTimeDiff;
+            }
+        }
+    }
+
+    /**
+     * Print final profiling information from native code
+     */
+    public native void jniClose();
+
+    @Override
+    public void close() {
+        if (doProfiling)
+            logger.info("Time spent in setup for JNI call : " + (pairHMMSetupTime * 1e-9));
+        super.close();
+        jniClose();
+    }
+
+    //Copied from http://frommyplayground.com/how-to-load-native-jni-library-from-jar 
+
```

Comment:
This link doesn't seem to resolve, I get a domain parking page.  I'm assuming this was posted under a license that makes it ok to include in our project?